src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -439,12 +438,12 @@ public void forceMerge(boolean flush) {
     public abstract void forceMerge(boolean flush, int maxNumSegments, boolean onlyExpungeDeletes, boolean upgrade) throws EngineException;
 
     /**
-     * Snapshots the index and returns a handle to it. Will always try and "commit" the
+     * Snapshots the index and returns a handle to it. If needed will try and "commit" the
      * lucene index to make sure we have a "fresh" copy of the files to snapshot.
+     *
+     * @param flushFirst indicates whether the engine should flush before returning the snapshot
      */
-    public abstract SnapshotIndexCommit snapshotIndex() throws EngineException;
-
-    public abstract void recover(RecoveryHandler recoveryHandler) throws EngineException;
+    public abstract SnapshotIndexCommit snapshotIndex(boolean flushFirst) throws EngineException;
```

Comment:
yeah, I can unbundle this and change those who use true to make a separate flush call, but we'll need a wait to not commit/open a new translog. I think we can probably always commit the translog now (one parameter less) but didn't want to go on that adventure now. Rather do it in another iteration.